qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
+        Tuple each selected '.cat' file and with same directory 'Wdf' file.
+
+        Collect all tuples in virtio_drivers list to be used signtool test.
+
+        :param filterd_cat_file:each filterd suffix '.cat' file
+        """
+        pattern = r'\w:\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        match_ret = re.match(pattern, filterd_cat_file)
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (filterd_cat_file, x), driver_sys)
+        cat_inf = map(lambda x: (filterd_cat_file, x), driver_inf)
+        cat_wdf = map(lambda x: (filterd_cat_file, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        disk_result = session.cmd_output(get_disk_letter)
+        disk_letter = re.search(r'(\w:)\s', disk_result, re.M).group(1)
+        drive_volume_list.append(disk_letter)
+
+        for drive in drive_volume_list:
+            for type in ['.cat', '.sys', '.inf', 'Wdf']:
                 driver = session.cmd_output(list_files_cmd % (drive, type)).\
                     splitlines()[1:-1]
                 drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
-                    raise error.TestFail(msg)
+
+            cat_list = filter(filter_all_tested_cat_files, drivers['.cat'])
+            map(match_cat_and_verified_file, cat_list)
+
+            for driver_file in virtio_drivers:
+                test_cmd = signtool_cmd % (driver_file[0], driver_file[1])
+                test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    test.fail(msg)
                 results_all += result
                 re_suc = "Number of files successfully Verified: ([0-9]*)"
                 try:
-                    suc_num = re.findall(re_suc, result)[0]
+                    suc_num = re.search(re_suc, result, re.M).group(1)
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    test.fail(msg)
```

Comment:
Is this step necessary? Sounds impossible to meet this situation: signtool check pass but no suc_num returned

Reply:
lijin, here code funcion is: when the result not include the "Number of files successfully Verified: ([0-9]*)", it will report msg = "Fail to get Number of files successfully Verified". The re_suc = "Number of files successfully Verified: ([0-9]*)" is a condition that judge the verify successfully.

Reply:
In my understanding, if result does not include  "Number of files successfully Verified: ([0-9])", the error will be caught in line 126 and whole script will be interrupted, it will never execute these lines.

Reply:
lijin, on 126 line, if status not 0, means the cmd cannot be executed completely, 
the 133 is under the cmd run completely, but the output not include the verify info, like some tool's error or print other none verify info, the status =0, but not means the verify successfully. this is an accurate screening condition.

Reply:
Could you have a double confirm?
From my side, if verify fail(I give wrong file to the cmd), then the status!=0, it will be caught by line126:
Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-58.0-py2.7.egg/avocado_vt/test.py:407
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-58.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)
  File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_driver_sign_check.py", line 133, in run
    test.fail(msg)
  File "/usr/lib/python2.7/site-packages/avocado_framework-58.0-py2.7.egg/avocado/core/test.py", line 1029, in fail
    raise exceptions.TestFail(message)
TestFail: Fail command: D:\signtool\Signtool_amd64_HCK.exe verify /v /kp /c E:\Balloon/2k12R2/amd64/balloon.cat E:\Balloon/2k12R2/amd64/balloon.pdb. Output: 
Verifying: E:\Balloon/2k12R2/amd64/balloon.pdb

Number of files successfully Verified: 0
Number of warnings: 0
Number of errors: 1
SignTool Error: File not found in the specified catalog.
SignTool Error: File not valid: E:\Balloon/2k12R2/amd64/balloon.pdb